src/traccuracy/matchers/_matched.py

```diff
@@ -1,45 +1,82 @@
-        and any additional private attributes that may be needed/used e.g. detection matrices.
+    Each Matcher subclass will implement its own kwargs as needed.
+    In use, the Matcher object will be initialized with kwargs prior to running compute_mapping
+    on a particular dataset
+    """
+
+    def compute_mapping(self, gt_graph: "TrackingGraph", pred_graph: "TrackingGraph"):
+        """Run the matching on a given set of gt and pred TrackingGraph and returns a Matched object
+        with a new copy of each TrackingGraph
 
         Args:
             gt_graph (TrackingGraph): Tracking graph object for the gt
             pred_graph (TrackingGraph): Tracking graph object for the pred
+
+        Returns:
+            matched (Matched): Matched data object
+
+        Raises:
+            ValueError: gt and pred must be a TrackingGraph object
         """
-        self.gt_graph = gt_graph
-        self.pred_graph = pred_graph
+        if not isinstance(gt_graph, TrackingGraph) or not isinstance(
+            pred_graph, TrackingGraph
+        ):
+            raise ValueError(
+                "Input data must be a TrackingData object with a graph and segmentations"
+            )
 
-        self.mapping = self.compute_mapping()
+        matched = self._compute_mapping(gt_graph, pred_graph)
 
         # Report matching performance
-        total_gt = len(self.gt_graph.nodes())
-        matched_gt = len({m[0] for m in self.mapping})
-        total_pred = len(self.pred_graph.nodes())
-        matched_pred = len({m[1] for m in self.mapping})
+        total_gt = len(matched.gt_graph.nodes())
+        matched_gt = len({m[0] for m in matched.mapping})
+        total_pred = len(matched.pred_graph.nodes())
+        matched_pred = len({m[1] for m in matched.mapping})
         logger.info(f"Matched {matched_gt} out of {total_gt} ground truth nodes.")
         logger.info(f"Matched {matched_pred} out of {total_pred} predicted nodes.")
 
-    @abstractmethod
-    def compute_mapping(self):
-        """Computes a mapping of nodes in gt to nodes in pred
+        return matched
 
-        The mapping must be a list of tuples, e.g. [(gt_node, pred_node)]
+    @abstractmethod
+    def _compute_mapping(self, gt_graph: "TrackingGraph", pred_graph: "TrackingGraph"):
+        """Computes a mapping of nodes in gt to nodes in pred and returns a Matched object
 
         Raises:
             NotImplementedError
         """
         raise NotImplementedError
+
+
+class Matched:
+    """Matched data class which stores TrackingGraph objects for gt and pred
+    and the computed mapping
+
+    Each TrackingGraph will be a new copy on the original object
+
+    Args:
+        gt_graph (TrackingGraph): Tracking graph object for the gt
+        pred_graph (TrackingGraph): Tracking graph object for the pred
+
+    """
+
+    def __init__(
+        self,
+        gt_graph: "TrackingGraph",
+        pred_graph: "TrackingGraph",
+        mapping: list[tuple[Any, Any]],
```

Comment:
The typing part of this line is causing the tests to fail in Python 3.8

@DragaDoncila @tlambert03 help please 😬 typing is not my strong suit

Reply:
before 3.9, `tuple` (the actual symbol itself representing the python type) was not subscriptable.  meaning there was no `__class_getitem__` on the `tuple` class, and you couldn't do `tuple[...]`.  There are two options to deal with that:

1. use `typing.Tuple` instead of `tuple`:
   ```python
   from typing import Tuple, List
   # ...
   mapping: List[Tuple[Any, Any]]
   ```
2. add `from __future__ import annotations` to the top of the file... then use the regular types.
    ```python
     from __future__ import annotations
     # ...
     mapping: list[tuple[Any, Any]]
     ```

    The implication of adding `from __future__ import annotations` is that python treats all of your type hints as strings, and does not attempt to evaluate them at runtime (so, it doesn't matter that tuple doesn't have a `__class_getitem__`, method since it won't be called at runtime anyway).  details in [PEP 563](https://peps.python.org/pep-0563/)

In general, I prefer the second option (`__future__.annotations`) since it makes for cleaner looking modern files.  However, there are a few cases where that can be problematic; namely, if you or any other libraries try to evaluate your type hints at runtime (such as pydantic, typer, fastAPI, or magicgui might do)... so put a pin in the general idea that you might run into this later